src/users/users.repository.ts

```diff
@@ -95,6 +96,28 @@ export class UsersRepository {
     }
   }
 
+  async changeUserPassword(
+    changePasswordDto: ChangePasswordDto,
+  ): Promise<void> {
+    const { userId, password } = changePasswordDto;
+    const salt = await bcrypt.genSalt();
+    const hashedPassword = await bcrypt.hash(password, salt);
+    try {
+      await this.prisma.user.update({
+        where: { id: userId },
+        data: { password: hashedPassword },
+      });
+    } catch (error) {
+      if (
+        error instanceof Prisma.PrismaClientKnownRequestError &&
+        error.code === 'P2025'
+      ) {
+        throw new NotFoundException(`User with ID ${userId} not found.`);
```

Comment:
i'm not sure if a repository should return an exception like this, maybe you can throw a more generic exception and handle it in the service 🤔